src/gamelogic/gamemodel/CardRange.cpp

```diff
@@ -7,47 +7,75 @@
 
 #include "CardRange.hpp"
 
-#include <stdexcept>
 #include <utility>
+#include <algorithm>
 
 namespace GameModel {
-    CardRange::CardRange(std::shared_ptr<Board> board, std::size_t number) : board(std::move(board)) {
-        std::size_t drawableCards = this->board->getCardPile()->size();
-        if (number > drawableCards) {
-            throw std::runtime_error(std::string("Cannot create card range of size ") + std::to_string(number)
-                                     + "! Game has only " + std::to_string(drawableCards) + " left!");
-        }
+    CardRange::CardRange(Board &gameBoard, const std::size_t number) : initialState(
+            std::vector<CardType>{gameBoard.getCardPile().crbegin(), gameBoard.getCardPile().crbegin() + number}),
```

Comment:
Du rufst hier effektiv den Ctor von std::vector auf im initialiser. Du kannst die Argumente einfach direkt weitergeben, also:
`initialState(gameBoard.getCardPile().crbegin(), ...)`
Hier ist aktuell leider das Problem, dass du eventuell out of bounds gehst. Das wurde vorher mit einem check und einer exception abgefangen. So haben wir halt leider undefiniertes Verhalten. Vielleicht ist deshalb doch die alte Version besser. Dann kann initial state halt leider doch nicht const, da du ihn dann im Body setzen musst.

Reply:
Ich habe den Check aber jetzt ins Environment eingebaut, also es wird bei drawNCards abgefragt, passt dass dann so?

Reply:
Ich finde, die Klasse sollte keinerlei undefiniertes Verhalten zulassen. Meinungen @aul12 ?

Reply:
Stimme ich dir zu. Im Zweifelsfall einfach eine Exception schmeißen, die Kosten dafür können wir tolerieren